sklearn/cluster/tests/test_k_medoids.py

```diff
@@ -0,0 +1,150 @@
+"""Testing for K-Medoids"""
+import numpy as np
+from scipy.sparse import csc_matrix
+
+from sklearn.cluster import KMedoids, KMeans
+from sklearn.datasets import load_iris
+from sklearn.metrics.pairwise import PAIRWISE_DISTANCE_FUNCTIONS
+from sklearn.metrics.pairwise import euclidean_distances
+from sklearn.utils.testing import assert_array_equal
+from sklearn.utils.testing import assert_equal, assert_raise_message
+from sklearn.utils.testing import assert_greater
+from sklearn.utils.testing import assert_allclose
+
+rng = np.random.RandomState(0)
+X = rng.rand(100, 5)
+
+
+def test_kmedoids_input_validation_and_fit_check():
+    # Invalid parameters
+    assert_raise_message(ValueError, "n_clusters should be a nonnegative "
+                                     "integer. 0 was given",
+                         KMedoids(n_clusters=0).fit, X)
+
+    assert_raise_message(ValueError, "n_clusters should be a nonnegative "
+                                     "integer. None was given",
+                         KMedoids(n_clusters=None).fit, X)
+
+    assert_raise_message(ValueError, "max_iter should be a nonnegative "
+                                     "integer. 0 was given",
+                         KMedoids(n_clusters=1, max_iter=0).fit, X)
+
+    assert_raise_message(ValueError, "max_iter should be a nonnegative "
+                                     "integer. None was given",
+                         KMedoids(n_clusters=1, max_iter=None).fit, X)
+
+    assert_raise_message(ValueError, "init needs to be one of the following: "
+                                     "['random', 'heuristic']",
+                         KMedoids(init=None).fit, X)
+
+    # Trying to fit 3 samples to 8 clusters
+    Xsmall = rng.rand(5, 2)
+    assert_raise_message(ValueError, "The number of medoids (8) must be less "
+                                     "than the number of samples 5.",
+                         KMedoids(n_clusters=8).fit, Xsmall)
+
+
+def test_kmedoids_fit_naive_with_all_pairwise_distance_functions():
+    for metric in PAIRWISE_DISTANCE_FUNCTIONS.keys():
```

Comment:
I don't really see much benefit in testing *al* pairwise distance functions. They are obviously all already tested with `pairwise_distances` and it would be very strange if we special-cased them, rather than relying on the existing tested code.

Reply:
@jnothman Do you recommend changing this test to just use `"euclidean"`?

Reply:
Test Euclidean and check that results change sensibly with one other